### ISSUE TITLE
'VariableManager' object has no attribute '_loader' from 'template' lookup plugin

### ISSUE THREAD

**Issue by dbeech** [ansible/ansible#64745]
##### SUMMARY

Using the "template" lookup function with a template containing hostvars causes the playbook execution to fail with error `'VariableManager' object has no attribute '_loader'`

The problem only exists in version 2.9.0. 
Version 2.8.6 and devel branch do not show the issue. 

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

Lookup plugin 'template' / VariableManager

##### ANSIBLE VERSION

```paste below
[root@227a4c00f1fe ~]# ansible --version
ansible 2.9.0
  config file = /root/.ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Jun 20 2019, 20:27:34) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]
```

##### CONFIGURATION

```paste below
[root@227a4c00f1fe ~]# ansible-config dump --only-changed
DEFAULT_SCP_IF_SSH(/root/.ansible.cfg) = True
HOST_KEY_CHECKING(/root/.ansible.cfg) = False
```

##### OS / ENVIRONMENT

CentOS 7.6 docker image
```
[root@227a4c00f1fe ~]# uname -a
Linux 227a4c00f1fe 4.9.184-linuxkit #1 SMP Tue Jul 2 22:58:16 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

[root@227a4c00f1fe ~]# cat /etc/redhat-release 
CentOS Linux release 7.6.1810 (Core)
```
##### STEPS TO REPRODUCE

```yaml
---
- name: test template lookup
  hosts: all
  tasks:
    - name: show template output
      debug:
        msg: "{{ lookup('template', 'mytemplate.j2') }}"
```

*mytemplate.j2*
```
{{ hostvars['localhost'].ansible_version.full }}
```

##### EXPECTED RESULTS

```
TASK [show template output] ***************************************************************************************************************************************
ok: [localhost] => {
    "msg": "2.9.0"
}
```

##### ACTUAL RESULTS

```paste below
[root@227a4c00f1fe ~]# ansible-playbook -vvvv -i hosts --connection=local myplaybook.yml 
ansible-playbook 2.9.0
  config file = /root/.ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 2.7.5 (default, Jun 20 2019, 20:27:34) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]
Using /root/.ansible.cfg as config file
setting up inventory plugins
host_list declined parsing /root/hosts as it did not pass its verify_file() method
script declined parsing /root/hosts as it did not pass its verify_file() method
auto declined parsing /root/hosts as it did not pass its verify_file() method
Set default localhost to localhost
Parsed /root/hosts inventory source with ini plugin
Loading callback plugin default of type stdout, v2.0 from /usr/lib/python2.7/site-packages/ansible/plugins/callback/default.pyc

PLAYBOOK: myplaybook.yml ******************************************************************************************************************************************
Positional arguments: myplaybook.yml
become_method: sudo
inventory: (u'/root/hosts',)
forks: 5
tags: (u'all',)
verbosity: 4
connection: local
timeout: 10
1 plays in myplaybook.yml

PLAY [test template lookup] ***************************************************************************************************************************************

TASK [Gathering Facts] ********************************************************************************************************************************************
task path: /root/myplaybook.yml:2
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: root
<localhost> EXEC /bin/sh -c 'echo ~root && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-tmp-1573582705.05-6277214451738 `" && echo ansible-tmp-1573582705.05-6277214451738="` echo /root/.ansible/tmp/ansible-tmp-1573582705.05-6277214451738 `" ) && sleep 0'
<localhost> Attempting python interpreter discovery
<localhost> EXEC /bin/sh -c 'echo PLATFORM; uname; echo FOUND; command -v '"'"'/usr/bin/python'"'"'; command -v '"'"'python3.7'"'"'; command -v '"'"'python3.6'"'"'; command -v '"'"'python3.5'"'"'; command -v '"'"'python2.7'"'"'; command -v '"'"'python2.6'"'"'; command -v '"'"'/usr/libexec/platform-python'"'"'; command -v '"'"'/usr/bin/python3'"'"'; command -v '"'"'python'"'"'; echo ENDFOUND && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/bin/python && sleep 0'
Using module file /usr/lib/python2.7/site-packages/ansible/modules/system/setup.py
<localhost> PUT /root/.ansible/tmp/ansible-local-844W7zX1_/tmpCISuzv TO /root/.ansible/tmp/ansible-tmp-1573582705.05-6277214451738/AnsiballZ_setup.py
<localhost> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1573582705.05-6277214451738/ /root/.ansible/tmp/ansible-tmp-1573582705.05-6277214451738/AnsiballZ_setup.py && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/bin/python /root/.ansible/tmp/ansible-tmp-1573582705.05-6277214451738/AnsiballZ_setup.py && sleep 0'
<localhost> EXEC /bin/sh -c 'rm -f -r /root/.ansible/tmp/ansible-tmp-1573582705.05-6277214451738/ > /dev/null 2>&1 && sleep 0'
ok: [localhost]
META: ran handlers

TASK [show template output] ***************************************************************************************************************************************
task path: /root/myplaybook.yml:5
File lookup using /root/mytemplate.j2 as file
fatal: [localhost]: FAILED! => {
    "msg": "An unhandled exception occurred while running the lookup plugin 'template'. Error was a <type 'exceptions.AttributeError'>, original message: 'VariableManager' object has no attribute '_loader'"
}

PLAY RECAP ********************************************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0  
```

**Comment by dbeech** [comment]
The issue looks similar to #57437 which was closed pending info.

**Comment by javery45** [comment]
I had this exact error message. After some digging I came across #62600 which was very similar to #63940, and in the comments others had agreed that it was basically a duplicate. The fix for #63940 was #64070, which I was able to manually apply to my Ansible in order to resolve this bug. As of writing this, I don't see a newer production release just yet that includes this fix, just development releases.

**Comment by Akasurde** [comment]
cc @mkrizek

**Comment by mkrizek** [comment]
This has been resolved in https://github.com/ansible/ansible/pull/64070 which will be included in the upcoming `2.9.1` release.

If you have further questions please stop by IRC or the mailing list:
* IRC: #ansible on irc.freenode.net
* mailing list: https://groups.google.com/forum/#!forum/ansible-project

**Comment by dbeech** [comment]
Thank you @javery45 and @mkrizek!
